src/main/java/com/google/sps/data/Election.java

```diff
@@ -132,8 +133,9 @@ public Election fromVoterInfoQuery(DatastoreService datastore, JSONObject voterI
       throws JSONException {
     Set<Long> contestKeyList = this.getContests();
     Set<Long> referendumKeyList = this.getReferendums();
-    ArrayList<EmbeddedEntity> pollingStations =
-        new ArrayList<EmbeddedEntity>(this.getPollingStations());
+
+    List<EmbeddedEntity> pollingStations = new ArrayList<EmbeddedEntity>(this.getPollingStations());
+    ImmutableSet<EmbeddedEntity> pollingStationSet;
```

Comment:
Do you know if the EmbeddedEntity hashcode/equals functions behave well for use in sets?

Sets depend on objects implementing equals() and hashcode(). Sometimes they will not behave as you expect (e.g. instead of checking that all the keys and values are the same, maybe they check that the variables literally refer to the same object, like pointer equality). 

See: go/ej3e-10

Reply:
according to this documentation (https://googleapis.dev/python/datastore/latest/entities.html), "Entities compare equal if their keys compare equal and their properties compare equal" using the __eq__ method. EmbeddedEntities are simply Entities intended to be stored as properties, so I'm assuming that the same functionality extends here, based on the description here (https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Embedded_entities).